xtask/src/test.rs

```diff
@@ -122,8 +123,32 @@ pub fn run_wasm() -> Result<()> {
         bail_on_err(&output, "Failed to install test dependencies")?;
     }
 
-    let output = Command::new(npm).arg("test").output()?;
-    bail_on_err(&output, &format!("Failed to run {npm} test"))?;
+    println!("Running {npm} test...");
+    let mut child = Command::new(npm)
+        .arg("test")
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()?;
+
+    // Display test results
+    if let Some(stdout) = child.stdout.take() {
+        let reader = io::BufReader::new(stdout);
+        for line in reader.lines() {
+            println!("{}", line?);
+        }
+    }
+
+    if let Some(stderr) = child.stderr.take() {
+        let reader = io::BufReader::new(stderr);
+        for line in reader.lines() {
+            eprintln!("{}", line?);
+        }
+    }
+
+    let status = child.wait()?;
+    if !status.success() {
+        anyhow::bail!("{npm} test failed");
+    }
```

Comment:
We have `bail_on_err`, so actually you could call `.output` to hide stdout/err by default, only print err and exit if there's an error, and then print stdout if it's fine. Look at `run_wasm_exports`